src/max_weight_matching.rs

```diff
@@ -0,0 +1,1307 @@
+    // label_ends[b] is the remote endpoint of the edge through which b
+    // obtained its label, or None if b's base vertex is single.
+    // If v is a vertex inside a T-blossom and label[v] == 2,
+    // label_ends[v] is the remote endpoint of the edge through which v is
+    // reachable from outside the blossom
+    let mut label_ends: Vec<Option<usize>>;
+    // If v is a vertex/node
+    // in_blossoms[v] is the top-level blossom to which v belongs.
+    // If v is a top-level vertex, v is itself a blossom (a trivial blossom)
+    // and in_blossoms[v] == v.
+    // Initially all nodes are top-level trivial blossoms.
+    let mut in_blossoms: Vec<usize> = (0..num_nodes).collect();
+    // if b is a sub-blossom
+    // blossom_parents[b] is its immediate parent
+    // If b is a top-level blossom, blossom_parents[b] is None
+    let mut blossom_parents: Vec<Option<usize>> = vec![None; 2 * num_nodes];
+    // If b is a non-trivial (sub-)blossom,
+    // blossom_children[b] is an ordered list of its sub-blossoms, starting with
+    // the base and going round the blossom.
+    let mut blossom_children: Vec<Vec<usize>> =
+        (0..2 * num_nodes).map(|_| Vec::new()).collect();
+    // If b is a (sub-)blossom,
+    // blossombase[b] is its base VERTEX (i.e. recursive sub-blossom).
+    let mut blossom_base: Vec<Option<usize>> =
+        (0..num_nodes).map(Some).collect();
+    blossom_base.append(&mut vec![None; num_nodes]);
+    // If b is a non-trivial (sub-)blossom,
+    // blossom_endpoints[b] is a list of endpoints on its connecting edges,
+    // such that blossom_endpoints[b][i] is the local endpoint of
+    // blossom_children[b][i] on the edge that connects it to
+    // blossom_children[b][wrap(i+1)].
+    let mut blossom_endpoints: Vec<Vec<usize>> =
+        (0..2 * num_nodes).map(|_| Vec::new()).collect();
+    // If v is a free vertex (or an unreached vertex inside a T-blossom),
+    // best_edge[v] is the edge to an S-vertex with least slack,
+    // or None if there is no such edge. If b is a (possibly trivial)
+    // top-level S-blossom,
+    // best_edge[b] is the least-slack edge to a different S-blossom,
+    // or None if there is no such edge.
+    let mut best_edge: Vec<Option<usize>>;
+    // If b is a non-trivial top-level S-blossom,
+    // blossom_best_edges[b] is a list of least-slack edges to neighboring
+    // S-blossoms, or None if no such list has been computed yet.
+    // This is used for efficient computation of delta3.
+    let mut blossom_best_edges: Vec<Vec<usize>> =
+        (0..2 * num_nodes).map(|_| Vec::new()).collect();
+    let mut unused_blossoms: Vec<usize> = (num_nodes..2 * num_nodes).collect();
+    // If v is a vertex,
+    // dual_var[v] = 2 * u(v) where u(v) is the v's variable in the dual
+    // optimization problem (multiplication by two ensures integer values
+    // throughout the algorithm if all edge weights are integers).
+    // If b is a non-trivial blossom,
```

Comment:
Commenting `v in 0..num_nodes` and `b in num_nodes..2 * num_nodes` somewhere may help readers a lot.

Reply:
Yep I agree, it's definitely a bit confusing without explaining the split in the values. I added this to the bottom of the comment in: https://github.com/Qiskit/retworkx/pull/229/commits/37b518da230c49a08f8173607f6bdb49f2fe5103